### PR TITLE
build(eslint-config-fluid): Update settings

### PR DESCRIPTION
**Pull request by tylerbutler**
The new version of typescript-eslint has new configs to extend from, which enable several new rules. This PR switches to those new configs and disables a few of the new rules we don't need to us right now. Changes implied by this PR are preapplied in #17380.

Summary of the new rules:

- [@typescript-eslint/ban-tslint-comment](https://typescript-eslint.io/rules/ban-tslint-comment/)
- [@typescript-eslint/class-literal-property-style](https://typescript-eslint.io/rules/class-literal-property-style/)
- [@typescript-eslint/consistent-generic-constructors](https://typescript-eslint.io/rules/consistent-generic-constructors/) (strict only)
- [@typescript-eslint/consistent-indexed-object-style](https://typescript-eslint.io/rules/consistent-indexed-object-style]/) (strict only)
- [@typescript-eslint/no-confusing-non-null-assertion](https://typescript-eslint.io/rules/no-confusing-non-null-assertion/)
- [@typescript-eslint/no-duplicate-enum-values](https://typescript-eslint.io/rules/no-duplicate-enum-values/)
- [@typescript-eslint/prefer-string-starts-ends-with](https://typescript-eslint.io/rules/prefer-string-starts-ends-with/)
- jsdoc/check-line-alignment (now a warning because of false positives in code blocks within TSDocs)
- unicorn/expiring-todo-comments (disabled because it seems to crash in some of our code, and we don't rely on the rule AFAIK)
- unicorn/no-unsafe-regex (disabled because [it's been deprecated](https://github.com/sindresorhus/eslint-plugin-unicorn/blob/6d15a02d48de7ecfc38d0683a8487b2f937d83a0/docs/deprecated-rules.md?plain=1#L19))

### CODE REVIEW

File: common/build/eslint-config-fluid/printed-configs/default.json [hunk 1]
```diff
@@ -908,9 +926,6 @@
             },
             "ForInStatement"
         ],
-        "no-return-await": [
```
Comment (Josmithr) [inline common/build/eslint-config-fluid/printed-configs/default.json:911]: This rule seems worth keeping - should we enable it explicitly?

File: common/build/eslint-config-fluid/printed-configs/default.json [hunk 2]
```diff
@@ -1353,9 +1367,6 @@
         "unicorn/no-unreadable-iife": [
             "error"
         ],
-        "unicorn/no-unsafe-regex": [
```
Comment (Josmithr) [inline common/build/eslint-config-fluid/printed-configs/default.json:1356]: Does this need to be disabled?

**Review by Josmithr — APPROVED** [review]
I left a couple of questions about rules that seem to have been removed that I think we may want to keep. Otherwise, looks good to me!

File: common/build/eslint-config-fluid/printed-configs/default.json (hunk 1 above)
Comment (tylerbutler) [reply common/build/eslint-config-fluid/printed-configs/default.json:911]: Good catch. It's superseded by https://typescript-eslint.io/rules/require-await/, which I will enable.

File: common/build/eslint-config-fluid/printed-configs/default.json (hunk 2 above)
Comment (tylerbutler) [reply common/build/eslint-config-fluid/printed-configs/default.json:1356]: It's now deprecated: https://github.com/sindresorhus/eslint-plugin-unicorn/blob/6d15a02d48de7ecfc38d0683a8487b2f937d83a0/docs/deprecated-rules.md?plain=1#L19

Comment (Josmithr) [reply common/build/eslint-config-fluid/printed-configs/default.json]: Probably worth adding a comment to the PR description.

File: common/build/eslint-config-fluid/base.js [hunk 3]
```diff
@@ -102,9 +102,10 @@ module.exports = {
 				avoidEscape: true,
 			},
 		],
+		"@typescript-eslint/require-await": "off",
 		"@typescript-eslint/restrict-plus-operands": "error",
 		"@typescript-eslint/restrict-template-expressions": "off",
-		"@typescript-eslint/require-await": "off",
+		"@typescript-eslint/return-await": "error",
```
Comment (alexvy86) [inline common/build/eslint-config-fluid/base.js:108]: (Post-merge comment) I think we should consider changing the default configuration for this one from `in-try-catch` to `always`. The stack trace benefit is worth it, IMO.

**Pull request merged**